### DISCUSSION TITLE
Is there a bug?

### DISCUSSION THREAD

**Discussion by wddpfx** [eProsima/Fast-DDS#4344] [General]
![image_18](https://github.com/eProsima/Fast-DDS/assets/19266720/2d65d67c-2e50-4b3e-bd86-f983f66329d8)

As dispalied in  th picture.
If we find the typeinformation in the map of "informations_“ by the key "min_identifier" ,why we should call the function of "fill_minimal_information"?  Is it OK just "information = innerInfo->second"?

**Comment by JLBuenoLopez** [comment]
@wddpfx 

Thanks for the post. The TypeObject type representation and related information defined in XTypes specification is being refactored in [feature/xtypes1.3](https://github.com/eProsima/Fast-DDS/tree/feature/xtypes1.3). This refactor is expected to be released in the next few months. The legacy code is going to be removed and it is not going to be fixed.

**Comment by wddpfx** [comment]
好的，收到了你的邮件，我会尽快查看。
